### PR TITLE
Create Gherkin tests for device-reachability-status-subscriptions & device-roaming-status-subscriptions

### PR DESCRIPTION
**Pull request by mdomale**
Gherkin tests for

device-reachability-status-subscriptions
device-roaming-status-subscriptions

#### What type of PR is this?

* tests 

#### What this PR does / why we need it:
Gherkin tests for device-reachability-status-subscriptions &
device-roaming-status-subscriptions

#### Which issue(s) this PR fixes:

Fixes #

#### Changelog input

```
 release-note

```

#### Additional documentation 

This section can be blank.

```
docs

```

### CODE REVIEW

**Comment by mdomale** [comment]
Updated as per new camara guidelines

**Comment by mdomale** [comment]
@bigludo7  @sachinvodafone  PR is updated aligned with comments from other PR's .Kindly review cc @akoshunyadi

File: code/Test_definitions/device-reachability-status-subscriptions.feature [hunk 1]
```diff
@@ -0,0 +1,225 @@
...
+Scenario: Receive notification for subscription-ends event on expiry  
+    Given that subscriptions are created synchronously
+    And a valid subscription request body
+    And the request body property "$.subscriptionExpireTime" is set to a value in the near future
+    When the request "createSubscription" is sent
+    Then the response code is 201 
+    Then the subscription is expired
+    Then event notification "subscription-ends" is received on callback-url
+    And notification body complies with the OAS schema at "##/components/schemas/EventSubscriptionEnds"
+    And type="org.camaraproject.geofencing-subscriptions.v0.subscription-ends"
```
Comment (bigludo7) [inline code/Test_definitions/device-reachability-status-subscriptions.feature:120]: The type should be:
`org.camaraproject.device-reachability-status-subscriptions.v0.subscription-ends`

File: code/Test_definitions/device-reachability-status-subscriptions.feature [hunk 2]
```diff
@@ -0,0 +1,225 @@
...
+    Given that subscriptions are created synchronously
+    And a valid subscription request body  
+    And the request body property "$.type" is "reachability-data"
+    And the request body property "$.subscriptionMaxEvents" is set to 1 
+    When the request "createSubscription" is sent
+    Then the response code is 201 
+    Then event notification "reachability-data" is received on callback-url
+    Then event notification "subscription-ends" is received on callback-url
+    And notification body complies with the OAS schema at "##/components/schemas/EventSubscriptionEnds"
+    And type="org.camaraproject.geofencing-subscriptions.v0.subscription-ends"
```
Comment (bigludo7) [inline code/Test_definitions/device-reachability-status-subscriptions.feature:134]: The type should be:
`org.camaraproject.device-reachability-status-subscriptions.v0.subscription-ends`

File: code/Test_definitions/device-reachability-status-subscriptions.feature [hunk 3]
```diff
@@ -0,0 +1,225 @@
...
+   Scenario: Receive notification for subscription-ends event on deletion 
+    Given that subscriptions are created synchronously
+    And a valid subscription request body  
+    When the request "createSubscription" is sent
+    Then the response code is 201 
+    When the request "deleteReachabilityStatusSubscription" is sent
+    Then the response code is 202 or 204	
+    Then event notification "subscription-ends" is received on callback-url
+    And notification body complies with the OAS schema at "##/components/schemas/EventSubscriptionEnds"
+    And type="org.camaraproject.geofencing-subscriptions.v0.subscription-ends"
```
Comment (bigludo7) [inline code/Test_definitions/device-reachability-status-subscriptions.feature:147]: The type should be:
`org.camaraproject.device-reachability-status-subscriptions.v0.subscription-ends`

File: code/Test_definitions/device-reachability-status-subscriptions.feature [hunk 4]
```diff
@@ -0,0 +1,225 @@
...
+    And the response property "$.code" is "INVALID_CREDENTIAL"
+    And the response property "$.message" contains a user friendly text
+	
+@reachability_status_subscription_17_invalid_access_token_type
+   Scenario: subscription creation with invalid access token type 
+    Given a valid subscription request body 
+    And the request property "$.accessTokenType" is not "bearer"
+    When the request "createSubscription" is sent
+    Then the response property "$.status" is 400
+    And the response property "$.code" is "INVALID_TOKEN"
```
Comment (bigludo7) [inline code/Test_definitions/device-reachability-status-subscriptions.feature:196]: For alignement with geofencing:
`And the response property "$.code" is "INVALID_TOKEN" or "INVALID_ARGUMENT"`

File: code/Test_definitions/device-roaming-status-subscriptions.feature [hunk 5]
```diff
@@ -0,0 +1,238 @@
...
+    And the response property "$.code" is "INVALID_CREDENTIAL"
+    And the response property "$.message" contains a user friendly text
+	
+@roaming_status_subscription_18_invalid_access_token_type
+   Scenario: subscription creation with invalid access token type 
+    Given a valid subscription request body 
+    And the request property "$.accessTokenType" is not "bearer"
+    When the request "createSubscription" is sent
+    Then the response property "$.status" is 400
+    And the response property "$.code" is "INVALID_TOKEN"
```
Comment (bigludo7) [inline code/Test_definitions/device-roaming-status-subscriptions.feature:209]: And the response property "$.code" is "INVALID_TOKEN" or "INVALID_ARGUMENT"

File: code/Test_definitions/device-roaming-status-subscriptions.feature [hunk 6]
```diff
@@ -0,0 +1,238 @@
...
+  Scenario: Receive notification for subscription-ends event on expiry  
+    Given that subscriptions are created synchronously
+    And a valid subscription request body
+    And the request body property "$.subscriptionExpireTime" is set to a value in the near future
+    When the request "createSubscription" is sent
+    Then the response code is 201 
+    And the subscription is expired
+    And event notification "subscription-ends" is received on callback-url
+    And notification body complies with the OAS schema at "##/components/schemas/EventSubscriptionEnds"
+    And type="org.camaraproject.geofencing-subscriptions.v0.subscription-ends"
```
Comment (bigludo7) [inline code/Test_definitions/device-roaming-status-subscriptions.feature:133]: The type should be `org.camaraproject.roaming-status-subscriptions.v0.subscription-ends`

File: code/Test_definitions/device-roaming-status-subscriptions.feature [hunk 7]
```diff
@@ -0,0 +1,238 @@
...
+    Given that subscriptions are created synchronously
+    And a valid subscription request body  
+    And the request body property "$.type" is "roaming_on"
+    And the request body property "$.subscriptionMaxEvents" is set to 1 
+    When the request "createSubscription" is sent
+    Then the response code is 201 
+    Then event notification "roaming_on" is received on callback-url
+    Then event notification "subscription-ends" is received on callback-url
+    And notification body complies with the OAS schema at "##/components/schemas/EventSubscriptionEnds"
+    And type="org.camaraproject.geofencing-subscriptions.v0.subscription-ends"
```
Comment (bigludo7) [inline code/Test_definitions/device-roaming-status-subscriptions.feature:147]: The type should be `org.camaraproject.roaming-status-subscriptions.v0.subscription-ends`

File: code/Test_definitions/device-roaming-status-subscriptions.feature [hunk 8]
```diff
@@ -0,0 +1,238 @@
...
+   Scenario: Receive notification for subscription-ends event on deletion 
+    Given that subscriptions are created synchronously
+    And a valid subscription request body  
+    When the request "createSubscription" is sent
+    Then the response code is 201 
+    When the request "deleteRoamingStatusSubscription" is sent
+    Then the response code is 202 or 204	
+    Then event notification "subscription-ends" is received on callback-url
+    And notification body complies with the OAS schema at "##/components/schemas/EventSubscriptionEnds"
+    And type="org.camaraproject.geofencing-subscriptions.v0.subscription-ends"
```
Comment (bigludo7) [inline code/Test_definitions/device-roaming-status-subscriptions.feature:160]: The type should be `org.camaraproject.roaming-status-subscriptions.v0.subscription-ends`

**Review by bigludo7 — COMMENTED** [review]
Thanks @mdomale @akoshunyadi 
This is very good.
Very very small comments linked to copy/paste from geofencing.

As soon as fixed I will approve

**Comment by mdomale** [comment]
@bigludo7 @sachinvodafone  PR updated kindly review .cc @akoshunyadi

**Review by bigludo7 — DISMISSED** [review]
LGTM now
Thanks @mdomale

File: code/Test_definitions/device-roaming-status-subscriptions.feature [hunk 9]
```diff
@@ -0,0 +1,238 @@
...
+    When the request "createSubscription" is sent
+    Then the response property "$.status" is 400
+    And the response property "$.code" is "INVALID_PROTOCOL"
+    And the response property "$.message" contains a user friendly text
+
+@roaming_status_subscription_17_invalid_credential_type
+   Scenario: subscription creation with invalid credential type
+    Given a valid subscription request body 
+    And the request property "$.credentialType" is not "ACCESSTOKEN"
+    When the request "createSubscription" is sent
```
Comment (akoshunyadi) [inline code/Test_definitions/device-roaming-status-subscriptions.feature:198]: The correct name is createDeviceRoamingStatusSubscription

File: code/Test_definitions/device-roaming-status-subscriptions.feature [hunk 10]
```diff
@@ -0,0 +1,238 @@
...
+    Then the response property "$.status" is 400
+    And the response property "$.code" is "INVALID_CREDENTIAL"
+    And the response property "$.message" contains a user friendly text
+	
+@roaming_status_subscription_18_invalid_access_token_type
+   Scenario: subscription creation with invalid access token type 
+    Given a valid subscription request body 
+    And the request property "$.accessTokenType" is not "bearer"
+    When the request "createSubscription" is sent
```
Comment (akoshunyadi) [inline code/Test_definitions/device-roaming-status-subscriptions.feature:207]: The correct name is createDeviceRoamingStatusSubscription

File: code/Test_definitions/device-roaming-status-subscriptions.feature [hunk 11]
```diff
@@ -0,0 +1,238 @@
...
+    Then the response property "$.status" is 400
+    And the response property "$.code" is "INVALID_TOKEN" or "INVALID_ARGUMENT"
+    And the response property "$.message" contains a user friendly text
+
+@roaming_status_subscription_19_invalid_credentials
+   Scenario: subscription creation with invalid credentials
+    Given a valid subscription request body 
+    And header "Authorization" token is set to invalid credentials
+    When the request "createSubscription" is sent
```
Comment (akoshunyadi) [inline code/Test_definitions/device-roaming-status-subscriptions.feature:216]: The correct name is createDeviceRoamingStatusSubscription

File: code/Test_definitions/device-roaming-status-subscriptions.feature [hunk 12]
```diff
@@ -0,0 +1,238 @@
...
+    And the response property "$.code" is "UNAUTHENTICATED"
+    And the response property "$.message" contains a user friendly text
+
+@roaming_status_subscription_20_invalid_inconsistent_access_token
+   Scenario: subscription creation with inconsistent access token for requested events subscription
+  # To test this, a token have to be obtained for a different device
+    Given a valid subscription request body 
+    And the request body property "$.device" is set to a valid testing device supported by the service
+    And header "Authorization" set to access token referring different device
+    When the request "createSubscription" is sent
```
Comment (akoshunyadi) [inline code/Test_definitions/device-roaming-status-subscriptions.feature:227]: The correct name is createDeviceRoamingStatusSubscription

File: code/Test_definitions/device-roaming-status-subscriptions.feature [hunk 13]
```diff
@@ -0,0 +1,238 @@
...
+    Then the response property "$.status" is 403
+    And the response property "$.code" is "SUBSCRIPTION_MISMATCH"
+    And the response property "$.message" contains a user friendly text
+
+@roaming_status_subscription_21_unknown_subscription_id
+   Scenario: Get subscription when subscription-id is unknown to the system 
+    Given the path parameter property "$.subscriptionId" is unknown to the system
+    When the request "retrieveRoamingStatusSubscription" is sent
```
Comment (akoshunyadi) [inline code/Test_definitions/device-roaming-status-subscriptions.feature:235]: The correct name is retrieveSubscription

File: code/Test_definitions/device-roaming-status-subscriptions.feature [hunk 14]
```diff
@@ -0,0 +1,238 @@
...
+    And type="org.camaraproject.roaming-status-subscriptions.v0.subscription-ends"
+    And the response property "$.terminationReason" is "MAX_EVENTS_REACHED"
+
+  @roaming_status_subscriptions_13_subscription_delete_event_validation
+   Scenario: Receive notification for subscription-ends event on deletion 
+    Given that subscriptions are created synchronously
+    And a valid subscription request body  
+    When the request "createSubscription" is sent
+    Then the response code is 201 
+    When the request "deleteRoamingStatusSubscription" is sent
```
Comment (akoshunyadi) [inline code/Test_definitions/device-roaming-status-subscriptions.feature:156]: The correct name is deleteSubscription

**Comment by mdomale** [comment]
@bigludo7  @akoshunyadi  PR updated

**Review by akoshunyadi — APPROVED** [review]

**Pull request merged**